components/compositing/webview.rs

```diff
@@ -2,19 +2,190 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-use std::collections::hash_map::Entry;
+use std::cell::RefCell;
+use std::collections::hash_map::{Entry, Keys, Values, ValuesMut};
 use std::collections::HashMap;
+use std::rc::Rc;
 
 use base::id::{PipelineId, WebViewId};
-use webrender_api::units::DeviceRect;
+use compositing_traits::SendableFrameTree;
+use fnv::FnvHashSet;
+use log::debug;
+use script_traits::AnimationState;
+use webrender_api::units::{DeviceRect, LayoutVector2D};
 
-#[derive(Debug, Default)]
-pub struct WebView {
-    pub pipeline_id: Option<PipelineId>,
+use crate::compositor::PipelineDetails;
+use crate::IOCompositor;
+
+pub(crate) struct WebView {
```

Comment:
Could we add more documentation about what the web view represents and why we need to group operations by it? Also, I'm wondering if it relates in any way to a concept from the spec(for example is there a relation between a webview and a top-level bc? Perhaps a webview is to rendering what a top-level bc is to a set of event-loops?). For someone like me just trying to understand this change, it's not clear how this fits into the existing architecture.

Reply:
`WebView` here doesn't relate to any specification-related thing. We have a `WebView` in the API layer which does have documentation. This thing in `compositing` is called `WebView` now, but just for legacy reasons. A future change will likely rename it to something like `WebViewRenderer`. It's the thing that handles rendering a single `WebView`, but only once we finish reorganizing the compositor around this idea.